zenoh/src/query.rs

```diff
@@ -279,6 +280,61 @@ impl<'a, 'b, Handler> GetBuilder<'a, 'b, Handler> {
         self.timeout = timeout;
         self
     }
+
+    /// By default, `get` will filter out
+    #[zenoh_core::unstable]
+    pub fn accept_replies(self, value: ReplyKeyExpr) -> Self {
+        let Self {
+            session,
+            selector,
+            target,
+            consolidation,
+            timeout,
+            handler,
+        } = self;
+        Self {
+            session,
+            selector: selector.and_then(|mut s| {
+                let any_selparam = s.parameter_index(_REPLY_KEY_EXPR_ANY_SEL_PARAM)?;
+                match (value, any_selparam) {
+                    (ReplyKeyExpr::Any, None) => {
+                        let s = s.parameters_mut();
+                        if !s.is_empty() {
+                            s.push('&')
+                        }
+                        s.push_str(_REPLY_KEY_EXPR_ANY_SEL_PARAM);
+                    }
+                    (ReplyKeyExpr::MatchingQuery, Some(index)) => {
+                        let s = s.parameters_mut();
+                        let mut start = index as usize;
+                        let pend = start + _REPLY_KEY_EXPR_ANY_SEL_PARAM.len();
+                        if start != 0 {
+                            start -= 1
+                        }
+                        match s[pend..].find('&') {
+                            Some(end) => std::mem::drop(s.drain(start..end + pend)),
+                            None => s.truncate(pend),
+                        }
```

Comment:
I believe it's worth adding a test to verify the correctness of the parameters truncation.